pandas/core/base.py

```diff
@@ -7,8 +7,10 @@
 from pandas.core import common as com
 import pandas.core.nanops as nanops
 import pandas.lib as lib
+from pandas.compat.func_defaults import TRANSPOSE_DEFAULTS
```

Comment:
the idea is that you can simply define a validation function and call it with and extraneous args, kwargs (you may need to have some of these return args as well, if they had to interpret them).

this will eliminate a reader having to understand anything about what you are doing, and hide all of the details to this function, where it is completely isolated.